site/src/scaffold/scripts/popover.ts

```diff
@@ -0,0 +1,33 @@
+export function initPopover(container: Element) {
+	const popovers = container.querySelectorAll('details');
+
+	if (popovers.length === 0) {
+		return;
+	}
+
+	popovers.forEach(p =>
+		p.addEventListener('keydown', event => {
+			if (event.key === 'Escape') {
+				if (p.hasAttribute('open')) {
+					p.removeAttribute('open');
+				}
+				p.focus();
+			}
+		})
+	);
+
+	container.addEventListener('click', event => {
+		const targetPopover =
+			event.target instanceof Element && (event.target.closest('details') as HTMLDetailsElement)
+				? (event.target.closest('details') as HTMLDetailsElement)
+				: null;
+
+		const otherPopovers = Array.from(popovers).filter(el => el !== targetPopover);
```

Comment:
One issue with this is that this will all work as the page loads, but we cannot assume that a `.popover` element might not be added to the DOM later. I think you'll have to querySelector popovers on keypress and on click.

Reply:
updated